### ISSUE TITLE
Push procedures

### ISSUE THREAD

**Issue by dpkoch** [rosflight/mavlink_c_library#2]
Hey @superjax, I guess I never really explained to you how this repository works. Since I've been using this setup for 3 years now I guess I just assumed you were familiar with it too... duh.

So this repo is a little different. It's under a type of version control, but not how we usually use it. The difference is that this is basically a dumping ground for compiled versions of the changes made in the `mavlink` repo. That means the true history of changes is kept in the `mavlink` repo, and each commit here is linked to a commit in that repo. The way that linking happens is through the commit messages generated by the [update_c_library.sh](https://github.com/BYU-MAGICC/mavlink/blob/master/scripts/update_c_library.sh) script in the `mavlink` repo. It automatically includes in the commit message the hash code of the `mavlink` repo commit to which that particular build is linked. So if you want to know what the changes were to a particular commit in this repo, then you look at the commit message, find the hash code, and lookup that commit in the `mavlink` repo.

The bottom line of all this is that the only thing that should be pushing commits directly to this repo is the update_c_library.sh script. There are instructions at the top that should walk you through how to set it up (the repository url's in the instructions are wrong, but I didn't want to change them in case we decide to do a pull request back into the mavlink/mavlink repo).

Sound good? Let me know if you have questions.

**Comment by superjax** [comment]
Got it.  Yeah, sorry about breaking it a few times.  haha.  It should be better from now on.
